### PR TITLE
add fix for #1466

### PR DESCRIPTION
**Pull request by nhachicha**
@realm/java  #1466

### CODE REVIEW

File: realm/src/androidTest/java/io/realm/RealmTest.java [hunk 1]
```diff
@@ -1862,6 +1862,66 @@ public Boolean call() throws Exception {
...
+                    signalBgFinished.countDown();
+                }
+            }
+        }.start();
+
+        callerThreadCompleted.countDown();
+        signalBgFinished.await();
+        if (threadAssertionError[0] != null)
+            throw threadAssertionError[0];
+    }
```
Comment (kneth) [inline realm/src/androidTest/java/io/realm/RealmTest.java]: `realm.close();`

File: realm/src/androidTest/java/io/realm/RealmTest.java [hunk 2]
```diff
@@ -1862,6 +1862,66 @@ public Boolean call() throws Exception {
...
+
+                    RealmResults<AllTypes> allBg = backgroundRealm.where(AllTypes.class).findAll();
+                    assertNotNull(allBg);
+                    assertEquals(2, allBg.size());
+                    assertEquals(42, allBg.get(0).getColumnLong());
+                    assertEquals(7, allBg.get(1).getColumnLong());
+
+                    backgroundRealm.close();
+
+
```
Comment (kneth) [inline realm/src/androidTest/java/io/realm/RealmTest.java]: Remove empty lines.

**Comment by kneth** [comment]
:+1: except for my two minor comments.

**Comment by beeender** [comment]
:+1:

**Pull request merged**